typescript/introduction/tsc-basics/tsc-classes-intro.md

```diff
@@ -0,0 +1,62 @@
+---
+author: kapnobatai136
+
+type: normal
+
+category: must-know
+
+aspects:
+  - introduction
+
+---
+
+# Classes Intro
+
+---
+## Content
+
+If you're not familiar with classes, these represent the building blocks of the object-oriented class-based programming approach. TypeScript allows you to use classes, and when the code is compiled it is compatbile with all the major broswers and platofrms. In this insight we will take a look at a simple class, with more in-depth information being given in a future mission.
+
+So, what is a class? To best explain it, let's take a look at an example:
+
+```ts
+class Animal {
+  name: string;
+  constructor(animalName: string) {
+    this.greeting = animalName;
+  }
+  sayHi() {
+    console.log("Hi, I'm a " + this.greeting);
+  }
+}
+
+let zebra = new Animal("zebra");
+zebra.sayHi();
+// Hi, I'm a zebra
+```
+
+Classes are pieces of re-usable code, and they are one of the reasons TypeScript is so great. You can use the `Animal` class to create as many animals as you want (we have created a `zebra` instance). Although this is a simple piece of code, you can see how useful this can get as the logic gets more complex.
```

Comment:
```suggestion
The benefit of classes is that they allow us to define data and behavior in one place, and then create many entities that use that data and follow that behavior as we see fit. In the example above, we used the `Animal` class to create a `zebra` but we could've also created a `tiger` and an `eagle` too. This allows us to have reusable code and not repeat ourselves.

Note that using classes comes with some drawbacks as well[3]
```

[3] 

OOP teaches us to organize data along with its associated behaviors into units called classes. Other classes can inherit this data/behaviors further and add their own. For example, a `Mammal` can extend an `Animal`. This can lead to programs with many levels of these inheritances links that are strict and hard to change. For an example, you can read about the [Diamond Problem](https://en.wikipedia.org/wiki/Multiple_inheritance#The_diamond_problem)